src/DI/ServiceLookup/ServiceProviderEngine.cs

```diff
@@ -51,7 +52,15 @@ protected ServiceProviderEngine(IEnumerable<ServiceDescriptor> serviceDescriptor
             if (callSite != null)
             {
                 _callback?.OnCreate(callSite);
-                return RealizeService(callSite);
+
+                var timer = DependencyInjectionEventSource.Log.RealizingService(callSite);
+
+                var instance = RealizeService(callSite);
+
+                // REVIEW: NEED TO REPLACE WITH ValueStopwatch!
+                DependencyInjectionEventSource.Log.RealizedService(callSite, instance, timer);
```

Comment:
Instance would always be `Func<ServiceProviderEngineScope, object>` it's not very useful to trace it's type.